runtime/src/accounts.rs

```diff
@@ -554,9 +564,18 @@ impl Accounts {
 
             let message = &tx.message();
             let acc = raccs.as_mut().unwrap();
-            for ((i, key), account) in message.account_keys.iter().enumerate().zip(acc.0.iter()) {
+            for ((i, key), (account, rent)) in message
+                .account_keys
+                .iter()
+                .enumerate()
+                .zip(acc.0.iter_mut().zip(acc.2.iter_mut()))
+            {
                 if message.is_writable(i) {
-                    accounts.push((key, account));
+                    if account.rent_epoch == 0 {
+                        account.rent_epoch = current_epoch;
+                        *rent += rent_collector.update(account);
```

Comment:
I'm wondering if we can make rent a single number for the TX instead of this vector...

Reply:
We could. What's the advantage in doing that? Additionally, we might lose insight on which account paid rent, and which didn't for current tx execution. Which is something we aren't using atm, but could be useful later on to track account balance.

Reply:
the advantage is simplification of the loadresult type

Reply:
done